saq/queue/postgres.py

```diff
@@ -484,7 +484,29 @@ async def iter_jobs(
 
     async def abort(self, job: Job, error: str, ttl: float = 5) -> None:
         job.error = error
-        await self.update(job, status=Status.ABORTING)
+
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                SQL(
+                    dedent(
+                        """
+                        SELECT status
+                        FROM {jobs_table}
+                        WHERE key = %(key)s
+                        FOR UPDATE
+                        """
+                    )
+                ).format(
+                    jobs_table=self.jobs_table,
+                ),
+                {
+                    "key": job.key,
+                },
+            )
+            result = await cursor.fetchone()
+            await self.update(job, status=Status.ABORTING, connection=conn)
+        if result == (Status.QUEUED,):
+            await job.finish(Status.ABORTED, error=error)
```

Comment:
so this is all done in a transaction right? even job.finish? why didn’t you add a test? is this code tested? why did you delete one?

Reply:
There's a test for abort but it didn't test this edge case. I've updated the test now. As for the deleted test, there's already a base test case that's testing batch, so it wasn't necessary. It was also wrong, which is what prompted this fix.